redfish-core/lib/pcie.hpp

```diff
@@ -91,6 +92,235 @@ static inline void
         std::array<const char*, 1>{pcieDeviceInterface});
 }
 
+/**
+ * @brief Fill PCIeDevice Status and Health based on PCIeSlot Link Status
+ * @param[in,out]   resp        HTTP response.
+ * @param[in]       linkStatus  PCIeSlot Link Status.
+ */
+static inline void fillPcieDeviceStatus(crow::Response& resp,
+                                        const std::string& linkStatus)
+{
+    if (linkStatus ==
+        "xyz.openbmc_project.Inventory.Item.PCIeSlot.Status.Operational")
+    {
+        resp.jsonValue["Status"]["State"] = "Enabled";
+        resp.jsonValue["Status"]["Health"] = "OK";
+        return;
+    }
+
+    if (linkStatus ==
+        "xyz.openbmc_project.Inventory.Item.PCIeSlot.Status.Degraded")
+    {
+        resp.jsonValue["Status"]["State"] = "Enabled";
+        resp.jsonValue["Status"]["Health"] = "Degraded";
```

Comment:
This isn't a valid Health value

Reply:
Wasn't sure the value; but found there are the ones:
```
        "Health": {
            "enum": [
                "OK",
                "Warning",
                "Critical"
            ],
```
Should we mark this as Critical or Warning?
Based on Brad's logic it does not matter since it checks the Health is OK or not...

Reply:
Yeah, let's go critical here 

            elif device_in_slot.status.state == "Enabled" and \
                    device_in_slot.status.health != "OK":
                row["Link Status"] = "degraded"

Reply:
Changed to Warning